extensions/control-plane/api/management-api/contract-negotiation-api/src/main/java/org/eclipse/edc/connector/api/management/contractnegotiation/transform/JsonObjectToContractRequestTransformer.java

```diff
@@ -52,31 +49,41 @@ public JsonObjectToContractRequestTransformer(Monitor monitor) {
                 .counterPartyAddress(counterPartyAddressOrConnectorAddress(jsonObject, context))
                 .protocol(transformString(jsonObject.get(PROTOCOL), context));
 
+        contractRequestBuilder.contractOffer(contractOffer(jsonObject, context));
+
+        var callbackAddress = jsonObject.get(CALLBACK_ADDRESSES);
+        if (callbackAddress != null) {
+            var addresses = new ArrayList<CallbackAddress>();
+            transformArrayOrObject(callbackAddress, CallbackAddress.class, addresses::add, context);
+            contractRequestBuilder.callbackAddresses(addresses);
+        }
+
+        return contractRequestBuilder.build();
+    }
+
+    private ContractOffer contractOffer(@NotNull JsonObject jsonObject, @NotNull TransformerContext context) {
         var policyJson = jsonObject.get(POLICY);
-        if (policyJson != null) {
-            var policy = transformObject(jsonObject.get(POLICY), Policy.class, context);
-            contractRequestBuilder.policy(policy);
+        if (policyJson != null && !policyJson.asJsonArray().isEmpty()) {
+            var policy = transformObject(policyJson, Policy.class, context);
+            var offerId = transformString(policyJson.asJsonArray().getJsonObject(0).get(ID), context);
```

Comment:
unfortunately our `Policy` object does not have an `id` field, because in fact the `ContractOffer` could be see as a `Policy` extension, I think extracting a dedicate json-to-contractoffer transformer will make everything cleared